libtock/max17205.c

```diff
@@ -12,25 +12,23 @@ static struct max17205_data result   = { .fired = false, .rc = 0, .value0 = 0, .
 static subscribe_upcall* user_upcall = NULL;
 
 // Internal callback for faking synchronous reads
-static void internal_user_upcall(int return_code,
+static void internal_user_upcall(int status,
                                  int value0,
                                  int value1,
                                  void* ud) {
 
   struct max17205_data* data = (struct max17205_data*) ud;
-  data->rc     = return_code;
+  data->rc     = tock_status_to_returncode(status);
   data->value0 = value0;
   data->value1 = value1;
   data->fired  = true;
 }
 
-static bool is_busy = false;
```

Comment:
These changes (removing the check of whether a pending upcall has been issued by the capsules) seem unrelated to the essence of this PR. Is this done just because this isn't typically done in libtock-c driver API implementations? I'm fine with leaving it in here, but I'm curious to know whether there's an actual requirement because of the type / wrapper changes of this PR to remove these checks.

Reply:
Nope. Just bad practice to store a copy of `is_busy` here (and won't work if there is more than one app anyway). Since I'm (likely) never going to go through every file in libtock-c again, decided to fix now rather than leave the bug.

Reply:
Thanks for fixing this!